BUILD.md

```diff
@@ -25,178 +10,304 @@ For security related issues please see [SECURITY.md](SECURITY.md).
+### Build the simulator
+
+The Multi edition firmware can be built as a simulator for linux-amd64. To build it, run:
 
 ```sh
-make jlink-flash-bootloader
+make simulator
 ```
 
-You need to install the [BitBox02 Python Library](#BitBox02-Python-library) before you can flash the built firmware.
+### Flash instructions
 
-Load the firmware by the bootloader (requires loading bootloader.bin by JLink, if not already loaded on the device):
+#### Connect J-Link probe
+
+Connect the J-Link probe to the debug pins on the BitBox02 prototype board. The
+pinout of the board and the Arm JTAG/SWD 10-pin connector can be seen in the
+table below.
+
+| Signal | Bitbox02 # | Arm JTAG/SWD # |
+| ------ | ---------- | -------------- |
+| VCC    | 1          | 1              |
+| CLK    | 2          | 4              |
+| GND    | 3          | 3, 5           |
+| DIO    | 4          | 2              |
+
+See [bitbox schematics](doc/bb02_v2.10_schematics.pdf) and [Arm JTAG/SWD
+interface](https://developer.arm.com/documentation/101636/0100/Debug-and-Trace/JTAG-SWD-Interface)
+
+Plug **both** the J-Link probe and the BitBox02 into the computer using USB. A
+USB hub can be used.
+
+#### Flash bootloader using J-Link
+
+Load the bootloader by JLink (requires `JLinkExe` in `$PATH`).
 
 ```sh
-make flash-dev-firmware
+make jlink-flash-bootloader
 ```
 
+> [!NOTE]
+> To flash a bootloader for a development device
+> `make jlink-flash-bootloader-development`.
+
+#### Flash firmware using J-Link
+
 Load the firmware by JLink:
 
 ```sh
 make jlink-flash-firmware
 ```
 
-### Build reference documentation (Doxygen)
+#### Flash firmware using bootloader and python cli client
 
-Dependencies:
+> [!TIP]
+> This method does not require a J-Link probe while developing.
+
+Install the [BitBox02 Python CLI client](#bitbox02-python-cli-client).
+
+Load the firmware through the bootloader:
 
 ```sh
-brew install graphviz doxygen
+make flash-dev-firmware
 ```
 
-Build:
+### Build reference documentation (Doxygen)
 
 ```sh
 make docs
 ```
 
 To view the results, open `build/docs/html/index.html` in a web browser.
 
-### BitBox02 Python library
+### Debugging
 
-There is a Python api library in `py/bitbox02`.
-
-Run `pip install -r py/requirements.txt` to install the deps (virtualenv recommended).
+#### Debugging using the simulator
 
-`make -C py/bitbox02` to generate the protobuf files.
-
-To kick off some api calls:
+Run it with:
 
 ```sh
-./py/send_message.py
+./build-build/bin/simulator
 ```
 
-### Unit tests
+This launches a server simulating the firmware. The send_message tool can connect to it with:
+
+    ./py/send_message.py --simulator
+
+If you choose to create a wallet by restoring a mnemonic, the simulator will automatically use this
+mnemonic:
+
+    boring mistake dish oyster truth pigeon viable emerge sort crash wire portion cannon couple enact box walk height pull today solid off enable tide
+
+
+#### Debugging using the J-Link probe and GDB
```

Comment:
would be useful to have some copy/paste code snippets ready for how to log stuff from C and from Rust:

`util_log()` and `::util::log::log!(...)`. 

An example for logging a hex string in C using the new helper function would also be helpful.